### DISCUSSION TITLE
Deploy SLO-Gen on AKS. Need recommendations on architecture and configuration.

### DISCUSSION THREAD

**Discussion by ppothepa24** [google/slo-generator#483] [Q&A]
Hi Team,

We plan on deploying SLO-Generator as a central repository/ service to calculate SLI/SLO on 1000's of Services at an Organization levek. We plan on deploying it on AKS ( Azure) ? What would be the architecture look like? As far as the configuration goes, what would you recommend for the shared configuration as well as the custom configuration file provided by these services. Needed some guidance..

Thanks.

**Comment by lvaylet** [comment]
Hi @ppothepa24,

Thanks for your interest in the SLO Generator.

The current documentation is mostly geared towards Google Cloud Platform. However, the same concepts apply to other public cloud providers. Even on-prem environments. The general idea is:

- The SLO Generator itself runs in API mode, as a container. I am definitely not an expert with Azure, but services like AKS and Container Apps seem like good candidates.
- The shared configuration file and the SLO definitions are usually stored (and possibly versioned) in an object storage service. So they would go to Azure Blob Storage. Maybe not in the same bucket, depending on your data governance and IAM model (who is supposed to do what  on which resource?). The shared configuration file can be managed by an Ops/SRE team, for example, and the SLO definitions can be managed directly by business teams in separate bucket(s). This is just an example that requires adjustment depending on your governance and IAM models.
- The SLO Generator is triggered via external requests scheduled on a regular basis. I am not sure what the equivalent of Cloud Scheduler on Azure is, but you just need something with a cron-like syntax that is able to call an HTTP endpoint. A Kubernetes CronJob could do the trick for example.

I do not have any visuals to share right now but it is something I can work on. The architecture would look a lot like https://github.com/terraform-google-modules/terraform-google-slo#slo-generator-any-monitoring-backend, with service names adjusted for Azure.

Does that make sense?
